jwql/instrument_monitors/common_monitors/dark_monitor.py

```diff
@@ -0,0 +1,988 @@
+
+        # Note that "self.query_table.run_monitor == True" below is
+        # intentional. Switching = to "is" results in an error in the query.
+        query = session.query(self.query_table).join(
+            sub_query,
+            and_(
+                self.query_table.aperture == self.aperture,
+                self.query_table.end_time_mjd == subq.c.maxdate,
+                self.query_table.run_monitor == True
+            )
+        ).all()
+
+        query_count = len(query)
+        if query_count == 0:
+            query_result = 57357.0  # a.k.a. Dec 1, 2015 == CV3
+            logging.info(("No query history for {}. Beginning search date will be set to {}."
+                         .format(self.aperture, query_result)))
+        elif query_count > 1:
+            raise ValueError('More than one "most recent" query?')
+        else:
+            query_result = query[0].end_time_mjd
+
+        return query_result
+
+    def noise_check(self, new_noise_image, baseline_noise_image, threshold=1.5):
+        """Create the ratio of the stdev (noise) image to a baseline
+        noise image. Pixels in the ratio image with values above
+        ``threshold`` will be marked as newly noisy.
+
+        Parameters
+        ----------
+        new_noise_image : numpy.ndarray
+            2D array containing the noise image from the new data
+
+        baseline_noise_image : numpy.ndarray
+            2D array containing the baseline noise image to compare
+            against the new noise image.
+
+        threshold : float
+            ``(new_noise_image / baseline_noise_image)`` ratio value
+            above which a pixel is considered newly noisey.
+
+        Returns
+        -------
+        noisy : tuple
+            Tuple (of lists) of x,y coordinates of newly noisy pixels
+        """
+
+        # Avoid divide by zeros
+        zeros = baseline_noise_image == 0.
+        baseline_noise_image[zeros] = 1.
+        new_noise_image[zeros] += 1.
+
+        ratio = new_noise_image / baseline_noise_image
+        noisy = np.where(ratio > threshold)
+
+        return noisy
+
+    def read_baseline_slope_image(self, filename):
+        """Read in a baseline mean slope image and associated standard
+        deviation image from the give fits file
+
+        Parameters
+        ----------
+        filename : str
+            Name of fits file to be read in
+
+        Returns
+        -------
+        mean_image : numpy.ndarray
+            2D mean slope image
+
+        stdev_image : numpy.ndarray
+            2D stdev image
+        """
+
+        try:
+            with fits.open(filename) as hdu:
+                mean_image = hdu['MEAN'].data
+                stdev_image = hdu['STDEV'].data
+            return mean_image, stdev_image
+        except (FileNotFoundError, KeyError) as e:
+            logging.warning('Trying to read {}: {}'.format(filename, e))
+
+    @log_fail
+    @log_info
```

Comment:
It might be worthwhile to add these decorators to the `__init__()` function, too, since then any log output there would also be send to the terminal. (Or at least that's how I think these decorators work...)